engine/src/parser/expression_tree.cpp

```diff
@@ -227,16 +227,8 @@ cudf::data_type infer_type_from_literal_token(const lexer::token & token) {
       return parsed_double == casted_float ? cudf::data_type{cudf::type_id::FLOAT32} : cudf::data_type{cudf::type_id::FLOAT64};
     } else {
       int64_t parsed_int64 = std::stoll(token_value);
-      return parsed_int64 > std::numeric_limits<int32_t>::max()
-              || parsed_int64 < std::numeric_limits<int32_t>::min()
-                ? cudf::data_type{cudf::type_id::INT64}
-                : parsed_int64 > std::numeric_limits<int16_t>::max()
-                  || parsed_int64 < std::numeric_limits<int16_t>::min()
-                      ? cudf::data_type{cudf::type_id::INT32}
-                      : parsed_int64 > std::numeric_limits<int8_t>::max()
-                        || parsed_int64 < std::numeric_limits<int8_t>::min()
-                          ? cudf::data_type{cudf::type_id::INT16}
-                          : cudf::data_type{cudf::type_id::INT8};
+      // as other SQL engines, defaults to int32
+      return cudf::data_type{cudf::type_id::INT32};
```

Comment:
I think it should be:
```
return parsed_int64 > std::numeric_limits<int32_t>::max() 
              || parsed_int64 < std::numeric_limits<int32_t>::min()
                   ? cudf::data_type{cudf::type_id::INT64} : cudf::data_type{cudf::type_id::INT32};
```
so that if the number is bigger than what can fit in an int32 its treated accordingly, and only otherwise do we just treat it as a default of int32

Reply:
Fortunately for us, Calcite is smart enough and if the literal value es larger than a `int32`, then it applies `CAST` to a `BIGINT` in its Relational Algebra, so for this query:

`query = "select 3255455842 as col_a from df"  `

The Relational Algebra is:
```
DEBUG: com.blazingdb.calcite.application.RelationalAlgebraGenerator - non optimized
LogicalProject(col_a=[3255455842:BIGINT])
  LogicalTableScan(table=[[main, df]])

DEBUG: com.blazingdb.calcite.application.RelationalAlgebraGenerator - optimized
LogicalProject(col_a=[3255455842:BIGINT])
  BindableTableScan(table=[[main, df]], projects=[[]], aliases=[[col_a]])
```